truetype/truetype.go

```diff
@@ -306,6 +314,141 @@ func (f *Font) parseMaxp() error {
 	return nil
 }
 
+func (f *Font) parseName() error {
+	if len(f.name) < 18 { // The name table is three uint16s + at least one Name Record.  Name Records are six uint16s.
+		return FormatError("name table too short")
+	}
+	if u16(f.name, 0) != 0 { // The first uint16 in the name table should be set to 0, per the spec.
+		return FormatError(fmt.Sprintf("invalid format specifier: %d", u16(f.name, 0)))
+	}
+
+	numNames := int(u16(f.name, 2)) // The number of Name Records is the second uint16 in the name table.
+	stringOffset := u16(f.name, 4)  // And the offset marking the beginning of the character data is the third int16.
+
+	if len(f.name) < int(stringOffset) {
+		return FormatError("string offset is larger than the name table")
+	}
+
+	const ( // The platform IDs in the name table.
+		PLATFORM_UNICODE   = 0
+		PLATFORM_MACINTOSH = 1
+		PLATFORM_MICROSOFT = 3
+	)
+
+	const ( // Encoding IDs for the Microsoft platform.
+		MICROSOFT_SYMBOL_CS  = 1
+		MICROSOFT_UNICODE_CS = 2
+		MICROSOFT_UCS_4      = 10
+	)
+
+	// Keep track of which records contain the name and style information.
+	var foundUnicode, foundAppleEnglish, foundAppleRoman, foundMicrosoft, foundApple [2]int
+	foundMicrosoftEnglish := false
+
+	for offset, i := 6, 0; i < numNames; i, offset = i+1, offset+12 {
+		platformID, encodingID, languageID, nameID := u16(f.name, offset), u16(f.name, offset+2), u16(f.name, offset+4), u16(f.name, offset+6)
+
+		switch nameID {
+		case 1, 2:
+			if u16(f.name, offset+8) == 0 { // Offset + 8 is the location of the string length.  If zero, skip it.
+				break
+			}
+			switch platformID {
+			case PLATFORM_UNICODE:
+				foundUnicode[nameID-1] = offset
+			case PLATFORM_MACINTOSH:
+				if languageID == 0 { // The language ID for English is zero.
+					foundAppleEnglish[nameID-1] = offset
+				} else if encodingID == 0 { // A zero encoding ID means Roman.
+					foundAppleRoman[nameID-1] = offset
+				}
+			case PLATFORM_MICROSOFT:
+				if foundMicrosoft[nameID-1] == 0 || (languageID&0x3FF) == 0x009 { // Prefer Microsoft fonts in English.
+					if encodingID == MICROSOFT_SYMBOL_CS || encodingID == MICROSOFT_UNICODE_CS || encodingID == MICROSOFT_UCS_4 {
+						foundMicrosoftEnglish = (languageID & 0x3FF) == 9
```

Comment:
The parens are unnecessary, and once you remove them, gofmt should emphasize that.